ext/mxnet/symbol.c

```diff
@@ -280,29 +287,57 @@ symbol_set_attributes_i(VALUE key, VALUE value, VALUE arg)
     value_cstr = StringValueCStr(value);
   }
 
-  CHECK_CALL(MXNET_API(MXSymbolSetAttr)(handle, key_cstr, value_cstr));
+  rb_ary_push(args->gc_guard, key);
+  args->keys[args->cursor] = key_cstr;
+
+  rb_ary_push(args->gc_guard, value);
+  args->vals[args->cursor] = value_cstr;
+
+  ++args->cursor;
 
   return ST_CONTINUE;
 }
 
-/* Sets an attribute of the symbol.
+/* Sets attributes of the symbol.
  *
- * For example.  `A.send :_set_attr(foo: 'bar')` adds the mapping `"{foo: bar}"`
- * to the symbol's attribute dictionary.
+ * For example: `sym.send(:set_attributes, foo: "bar")` adds the
+ * mapping `{foo: "bar"}` to the symbol's attribute dictionary.
  *
  * @param kwargs [Hash{Symbol => #to_str}] The attributes to set
  */
 static VALUE
 symbol_set_attributes(int argc, VALUE *argv, VALUE obj)
 {
-  VALUE kwargs;
   SymbolHandle handle;
+  struct set_attributes_args args;
+  VALUE kwargs, gc_guard = Qnil, keys_str = Qnil, vals_str = Qnil;
+  char const **keys = NULL, **vals = NULL;
+  int num_attrs = 0;
```

Comment:
It should be `nn_uint` (or `mx_uint`) instead of `int`.

Reply:
fixed in c5227f7